source/api/ut.pkb

```diff
@@ -267,13 +279,16 @@ create or replace package body ut is
       a_client_character_set
     );
     if l_reporter is of (ut_output_reporter_base) then
-      l_lines := treat(l_reporter as ut_output_reporter_base).get_lines_cursor();
+      l_data := treat(l_reporter as ut_output_reporter_base).get_lines_cursor();
       loop
-        fetch l_lines into l_line;
-        exit when l_lines%notfound;
-        pipe row(l_line);
+        fetch l_data into l_clob, l_item_type;
+        exit when l_data%notfound;
+        l_lines := ut_utils.clob_to_table(l_clob, ut_utils.gc_max_storage_varchar2_len);
+        for i in 1 .. l_lines.count loop
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Replace this FOR loop by a more robust WHILE one. ([plsql:ForLoopUsageCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3AForLoopUsageCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=809&issues=AWgA5Ioj_MPaBDn0kbLv&open=AWgA5Ioj_MPaBDn0kbLv)

[ISSUE_KEY]: AWgA5Ioj_MPaBDn0kbLv